scripts/schemas/patch-schema.yml

```
@@ -0,0 +1,115 @@
+# Copyright (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# A pykwalify schema for basic validation of the patches.yml format.
+
+# The schema for individual patch objects
+schema;patch-schema:
+  type: seq
+  sequence:
+    - type: map
+      mapping:
+
+        # The path to the patch file, relative to the root of the module
+        # E.g. zephyr/kernel-pipe-fix-not-k-no-wait-and-ge-min-xfer-bytes.patch
+        path:
+          required: true
+          type: str
+
+        # The SHA-256 checksum of the patch file
+        # e.g. e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+        sha256sum:
+          required: true
+          type: str
+          pattern: "^[0-9a-f]{64}$"
+
+        # The path of the module the patch is for, relative to the west workspace
+        # e.g. zephyr, or bootloader/mcuboot
+        module:
+          required: true
+          type: str
+
+        # The name of the primary author of the patch, e.g. Kermit D. Frog
+        author:
+          required: true
+          type: str
+
+        # The email address of the primary author of the patch, e.g. itsnoteasy@being.gr
+        email:
+          required: true
+          type: str
+          pattern: ".+@.+"
+
+        # The date the patch was created, in ISO 8601 date format YYYY-MM-DD
+        date:
+          required: true
+          type: date
+          format: "%Y-%m-%d"
+
+        # Whether the patch should be submitted upstream
+        upstreamable:
+          type: bool
+          default: true
+
+        # The URL of the upstream pull request to merge the patch
+        # e.g. https://github.com/zephyrproject-rtos/zephyr/pull/24486
+        merge-pr:
+          type: str
+          pattern: "^https?://"
+
+        # The URL of the upstream issue associated with the patch, such as an enhancement issue
+        # or bug report, e.g. https://github.com/zephyrproject-rtos/zephyr/issues/24485
+        issue:
+          type: str
+          pattern: "^https?://"
+
+        # Whether the associated merge-pr has been merged
+        merge-status:
+          type: bool
+
+        # The upstream commit incorporating the associated merge-pr
+        # e.g. af926ae728c78affa89cbc1de811ab4211ed0f69, or af926ae728c
+        merge-commit:
+          type: str
+          pattern: "^([0-9a-f]{11}|[0-9a-f]{40})$"
```

Why restrict to only two forms? Any sha length is accepted for `git` if it uniquely identifies the commit.

Maybe make it a range
```suggestion
          pattern: "^[0-9a-f]{10,40}$"
```

---

Git default for short SHA is 7 digits, fwiw

---

Hmmm.. maybe the full sha is the simplest 🤷🏼‍♂️

---

I switched to only allow the 40-digit SHA-1 hash. Should be safe enough for now.